### PR TITLE
Rubocop friendly cops part 10

### PR DESCRIPTION
**Pull request by DavidKang**
@openSUSE/open-build-service Please review

The affected cops are:
* http://rubocop.readthedocs.io/en/latest/cops_style/#styleemptyliteral
* http://rubocop.readthedocs.io/en/latest/cops_style/#styleemptymethod
* http://rubocop.readthedocs.io/en/latest/cops_style/#stylehashsyntax
* http://rubocop.readthedocs.io/en/latest/cops_style/#styleencoding
* http://rubocop.readthedocs.io/en/latest/cops_style/#styleifunlessmodifier

Pair-programmed with: @bgeuken

### CODE REVIEW

**Comment by mdeniz** [comment]
I created a separate PR (https://github.com/openSUSE/open-build-service/pull/4265) for having master without errors. Please remove your last commit, now it will be unneeded.

**Comment by mdeniz** [comment]
LGTM overall, but I have some doubts about the Fix Style/LineEndConcatenation cop, is what we want?

File: src/api/app/controllers/source_controller.rb [hunk 1]
```diff
@@ -302,9 +302,9 @@ class NoMatchingReleaseTarget < APIException
   def verify_can_modify_target_package!
     return if User.current.can_modify_package?(@package)
 
-    raise CmdExecutionNoPermission, "no permission to execute command '#{params[:cmd]}' " +
+    raise CmdExecutionNoPermission, "no permission to execute command '#{params[:cmd]}' " \
```
Comment (bgeuken) [inline src/api/app/controllers/source_controller.rb]: I can live with this change, but I don't really understand it... I need to look up the reasoning for why a backslash is preferred here.

**Comment by DavidKang** [comment]
@mdeniz, thanks 😸

File: src/api/app/controllers/webui/project_controller.rb [hunk 2]
```diff
@@ -1004,16 +1001,12 @@ def check_devel_package_status(currentpack, p)
...
-    end
+    currentpack['requests_to'].concat(@submits[key]) if @submits.has_key? key
 
     currentpack['develmd5'] = dp.verifymd5
     currentpack['develmtime'] = dp.maxmtime
 
-    if dp.error
-      currentpack['problems'] << 'error-' + dp.error
-    end
+    currentpack['problems'] << 'error-' + dp.error if dp.error
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/project_controller.rb]: [nitpick]
Guess it is autocorrected but what about string interpolation?
```"error-#{dp.error}"```

File: src/api/app/controllers/webui/project_controller.rb [hunk 3]
```diff
@@ -1004,16 +1001,12 @@ def check_devel_package_status(currentpack, p)
     currentpack['develproject'] = dproject
     currentpack['develpackage'] = dp.name
     key = "#{dproject}/#{dp.name}"
-    if @submits.has_key? key
-      currentpack['requests_to'].concat(@submits[key])
-    end
+    currentpack['requests_to'].concat(@submits[key]) if @submits.has_key? key
```
Comment (ChrisBr) [inline src/api/app/controllers/webui/project_controller.rb]: [nitpick]
Guess it is autocorrected but what about parentheses?
```@submits.has_key?(key)```

(https://github.com/bbatsov/rubocop/issues/3224 ``has_key?`` seems to be "deprecated")

File: src/api/app/helpers/webui/webui_helper.rb [hunk 4]
```diff
@@ -63,9 +63,7 @@ def format_projectname(prjname, login)
       else
         prjname = '~' + splitted[1]
       end
-      if splitted.length > 2
-        prjname += ':' + splitted[-1]
-      end
+      prjname += ':' + splitted[-1] if splitted.length > 2
```
Comment (ChrisBr) [inline src/api/app/helpers/webui/webui_helper.rb]: [nitpick]
String interpolation?

File: src/api/app/models/bs_request.rb [hunk 5]
```diff
@@ -173,9 +173,7 @@ def self.new_from_xml(xml)
       request.number = theid if theid
 
       actions = hashed.delete('action')
-      if actions.is_a? Hash
-        actions = [actions]
-      end
+      actions = [actions] if actions.is_a? Hash
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: [nitpick]
Parentheses?

**Pull request merged**